crates/ruff/tests/integration_test.rs

```diff
@@ -1523,7 +1581,17 @@ fn check_hints_hidden_unsafe_fixes() {
     exit_code: 1
     ----- stdout -----
     -:1:1: RUF901 [*] Hey this is a stable test rule with a safe fix.
+    Suggested fix:
```

Comment:
It looks like we need to do something special for violations without help messages? We want a dedicated test case for that and a test case for violations without source context.

Reply:
Well, for example, there's no source context for this test rule right? I think if the violation is attached to an empty range we elide the source context. This behavior was recently modified in https://github.com/astral-sh/ruff/pull/12304

Reply:
I'm going to open a new issue tracking what to do in each of those cases, with the existing baseline implementation here as a useful starting point.